### ISSUE TITLE
Aquamacs hangs when uncommenting in xml file

### ISSUE THREAD

**Issue by HayoBaan** [aquamacs-emacs/aquamacs-emacs#231]
When uncommenting a line in an xml file, Aquamacs hangs.

To reproduce, copy this text to a buffer in xml-mode:
```
<set>
  <!-- <xx>aaa</xx> -->
  <yy>aaa</yy>
</set>
```
Now hen you remove the comment (e.g. by using `uncomment-region` or by simply removing `<!--`), Aquamancs hangs.

**Comment by gnurser** [comment]
I've had this problem as well. it's a real pain. I believe it may be an issue with emacs-25, so emacs-26 versions do not have this issue.

**Comment by nega0** [comment]
_worksforme_ Aquamacs 3.4 & 3.6

are you using the default `nxml-mode`?

**Comment by HayoBaan** [comment]
> _worksforme_ Aquamacs 3.4 & 3.6

Interesting!

> are you using the default `nxml-mode`?

AFAIK, I'm using the default, `nXML Valid Fly Spc` in the mode-line.

**Comment by HayoBaan** [comment]
Note: I just tried disabling validation, but even with that disabled, it still hangs after removing the comment start ☹️

**Comment by nega0** [comment]
I was able to force it to hang. Looks like it's getting stuck in an infinite loop, and based on the functions, in [alloc.c](https://github.com/emacs-mirror/emacs/blob/master/src/alloc.c). If it's an Emacs issue as @gnurser suggests there probably isn't much to be done other than trying a newer version of nXML or waiting for @treese to start pushing out Aquamacs 4 builds.

it's a bit brute force, but maybe wrapping un/comment-region so that it drops to fundamental-mode, does its thing, then goes back to nXML will help. hacky, but better than hanging.

**Comment by HayoBaan** [comment]
> it's a bit brute force, but maybe wrapping un/comment-region so that it drops to fundamental-mode, does its thing, then goes back to nXML will help. hacky, but better than hanging.

Actually that's quite a good (manual) work-around for the infrequent times I actually need this. (I now fire-up vi, but temporarily switching to fundamental-mode is actually a bit easier). 👍

**Comment by treese** [comment]
Yes this is pretty annoying. Unfortunately, nxml-mode got an overhaul shortly after the current core Emacs code in Aquamacs was last updated. The bug is fixed in there somewhere, but it's not one easily pulled over as a patch, and the code has changed quite a bit.

So we'll pick up the fix in 4.0. I'll leave this open to check then.

**Comment by treese** [comment]
Also: there's some indication in the old emacs discussion of this bug that it's related to having flyspell also turned on, which may explain some of the "works for me" comments as well as give another workaround

**Comment by rpgoldman** [comment]
Sorry if this is a dumb question, but does this mean we could just install a local copy of nxml-mode with the package manager as a way to fix this?

**Comment by treese** [comment]
Not a dumb question at all! However, nxml-mode built in to Emac; it's not available as a separate package. It's possible that you could copy the lisp/nxml-mode from the Emacs source code, but it's quite likely that it depends on other changes in Emacs that aren't currently in Aquamacs (yet).

**Comment by HayoBaan** [comment]
> Also: there's some indication in the old emacs discussion of this bug that it's related to having flyspell also turned on, which may explain some of the "works for me" comments as well as give another workaround

Just to confirm disabling fly spell minor mode indeed works! So until this is permanently fixed, this is actually the best work-around!
